Derivatives of input no fatal error

### Description

When translating a model where an input needs to be differentiated omc reports an error but won't stop the translation. Instead it fails to compile the garbage C code later on.

### Steps to Reproduce

```Modelica
model illegalDerInput
  input Real u;
  output Real y;
equation
  y = der(u);
end illegalDerInput;
```

```
loadFile("illegalDerInput.mo"); getErrorString();
buildModel(illegalDerInput); getErrorString();
```

Error message:
```bash
Error: The model requires derivatives of some inputs as listed below:
der(u)
[...]
illegalDerInput_06inz.c:20:56: error: use of undeclared identifier '_omcQ_24DER'
  (data->localData[0]->realVars[1] /* y variable */) = _omcQ_24DER._u;
                                                       ^
1 error generated.
```

### Expected Behavior

Brake at the first reported error and don't continue.


### Version and OS

<!--- Please complete the following information. -->

 - OpenModelica Version: v1.21.0-dev-77-gf539850045-cmake
 - OS: Ubuntu 20.04.5 LTS

> Isn't this maybe too restrictive? 

I was a bit imprecise. I mean top-level inputs.
But top-level inputs from a CSV file seem to be an issue.
Try simulating `illegalDerInput` with `-simflags=input.csv` for `time^2` inputs:

```CSV
"time","illegalDerInput.u"
0,0
0.002,4e-06
0.004,1.6e-05
0.006,3.6e-05
0.008,6.4e-05
0.01,0.0001
0.012,0.000144
[...]
```

Fixing this is outside the scope of this issue, but probably a good idea.

But for now getting translation error

```
Error: The model requires derivatives of some inputs as listed below:
der(u)
Error: post-optimization module inputDerivativesUsed (simulation) failed.
```

is better than the C compilation error:
```
[...]
illegalDerInput.c:106:56: error: use of undeclared identifier '_omcQ_24DER'
  (data->localData[0]->realVars[1] /* y variable */) = _omcQ_24DER._u;
                                                       ^
illegalDerInput_06inz.c:20:56: error: use of undeclared identifier '_omcQ_24DER'
  (data->localData[0]->realVars[1] /* y variable */) = _omcQ_24DER._u;
                                                       ^
1 error generated.
1 error generated.
mingw32-make: *** [<builtin>: illegalDerInput_06inz.o] Error 1
mingw32-make: *** Waiting for unfinished jobs....
mingw32-make: *** [<builtin>: illegalDerInput.o] Error 1
mingw32-make: Leaving directory 'C:/Users/AHEUER~1/AppData/Local/Temp/OPENMO~1/OMEdit/ILLEGA~1'
Compilation process failed. Exited with code 0x00000002.
```

--- a/OMCompiler/Compiler/BackEnd/BackendDAEUtil.mo
+++ b/OMCompiler/Compiler/BackEnd/BackendDAEUtil.mo
@@ -9367,6 +9367,7 @@ algorithm
 end setFunctionTree;
 
 public function setEqSystEqs
+  "Set ordered equations of input equation system to given equations."
   input BackendDAE.EqSystem inSyst;
   input BackendDAE.EquationArray inEqs;
   output BackendDAE.EqSystem syst = inSyst;

--- a/OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo
+++ b/OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo
@@ -465,7 +465,7 @@ protected function inputDerivativesUsedWork "author: Frenkel TUD 2012-10"
   output BackendDAE.Shared outShared = inShared "unused";
   output Boolean outChanged;
 algorithm
-  (osyst, outChanged) := matchcontinue(isyst)
+  (osyst, outChanged) := match(isyst)
     local
       BackendDAE.EquationArray orderedEqs;
       list<DAE.Exp> explst;
@@ -474,10 +474,10 @@ algorithm
       ((_, explst as _::_)) = BackendDAEUtil.traverseBackendDAEExpsEqns(orderedEqs, traverserinputDerivativesUsed, (BackendVariable.daeGlobalKnownVars(inShared), {}));
       s = stringDelimitList(List.map(explst, ExpressionDump.printExpStr), "\n");
       Error.addMessage(Error.DERIVATIVE_INPUT, {s});
-    then (BackendDAEUtil.setEqSystEqs(isyst, orderedEqs), true);
+    then fail();
 
     else (isyst, inChanged);
-  end matchcontinue;
+  end match;
 end inputDerivativesUsedWork;
 
 protected function traverserinputDerivativesUsed "author: Frenkel TUD 2012-10"

--- a/OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo
+++ b/OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo
@@ -464,8 +464,10 @@ protected function inputDerivativesUsedWork "author: Frenkel TUD 2012-10"
   output BackendDAE.EqSystem osyst;
   output BackendDAE.Shared outShared = inShared "unused";
   output Boolean outChanged;
+protected
+  Boolean hasFailed = false;
 algorithm
-  (osyst, outChanged) := match(isyst)
+  (osyst, outChanged) := matchcontinue(isyst)
     local
       BackendDAE.EquationArray orderedEqs;
       list<DAE.Exp> explst;
@@ -474,10 +476,15 @@ algorithm
       ((_, explst as _::_)) = BackendDAEUtil.traverseBackendDAEExpsEqns(orderedEqs, traverserinputDerivativesUsed, (BackendVariable.daeGlobalKnownVars(inShared), {}));
       s = stringDelimitList(List.map(explst, ExpressionDump.printExpStr), "\n");
       Error.addMessage(Error.DERIVATIVE_INPUT, {s});
-    then fail();
+      hasFailed = true;
+    then (BackendDAEUtil.setEqSystEqs(isyst, orderedEqs), true);
 
     else (isyst, inChanged);
-  end match;
+  end matchcontinue;
+
+  // Fail after error is displayed.
+  // We do it this way, because I was to lazy to rewrite all of this function.
+  if hasFailed then fail(); end if;
 end inputDerivativesUsedWork;
 
 protected function traverserinputDerivativesUsed "author: Frenkel TUD 2012-10"

